domain/src/main/kotlin/no/nav/su/se/bakover/domain/vedtak/GjeldendeVedtaksdata.kt

```diff
@@ -60,41 +61,33 @@ data class GjeldendeVedtaksdata(
-                        )
-                    }
+                        it.vilkårsvurderinger is VilkårsvurderingerSøknadsbehandling.Uføre || it.vilkårsvurderinger is VilkårsvurderingerRevurdering.Uføre
+                    } -> VilkårsvurderingerRevurdering.Uføre(
+                        uføre = it.uføreVilkår(),
+                        lovligOpphold = it.lovligoppholdVilkår(),
+                        formue = it.formueVilkår(),
+                        utenlandsopphold = it.utenlandsoppholdVilkår(),
+                        opplysningsplikt = it.opplysningspliktVilkår(),
+                        flyktning = it.flyktningVilkår(),
+                        fastOpphold = it.fastOppholdINorgeVilkår(),
+                        personligOppmøte = it.personligOppmøteVilkår(),
+                        institusjonsopphold = it.institusjonsoppholdVilkår(),
+                    )
 
                     vedtakPåTidslinje.all {
-                        it.vilkårsvurderinger is Vilkårsvurderinger.Søknadsbehandling.Alder ||
-                            it.vilkårsvurderinger is Vilkårsvurderinger.Revurdering.Alder
-                    } -> {
-                        Vilkårsvurderinger.Revurdering.Alder(
-                            lovligOpphold = it.lovligoppholdVilkår(),
-                            formue = it.formueVilkår(),
-                            utenlandsopphold = it.utenlandsoppholdVilkår(),
-                            opplysningsplikt = it.opplysningspliktVilkår(),
-                            pensjon = it.pensjonsVilkår(),
-                            familiegjenforening = it.familiegjenforeningvilkår(),
-                            fastOpphold = it.fastOppholdINorgeVilkår(),
-                            personligOppmøte = it.personligOppmøteVilkår(),
-                        )
-                    }
-
-                    else -> {
-                        throw IllegalStateException("Kan ikke hente gjeldende vedtaksdata for blanding av uføre og alder.")
-                    }
+                        it.vilkårsvurderinger is VilkårsvurderingerSøknadsbehandling.Alder || it.vilkårsvurderinger is VilkårsvurderingerRevurdering.Alder
+                    } -> VilkårsvurderingerRevurdering.Alder(
+                        lovligOpphold = it.lovligoppholdVilkår(),
+                        formue = it.formueVilkår(),
+                        utenlandsopphold = it.utenlandsoppholdVilkår(),
+                        opplysningsplikt = it.opplysningspliktVilkår(),
+                        pensjon = it.pensjonsVilkår(),
+                        familiegjenforening = it.familiegjenforeningvilkår(),
+                        fastOpphold = it.fastOppholdINorgeVilkår(),
+                        personligOppmøte = it.personligOppmøteVilkår(),
+                    )
+
+                    else -> throw IllegalStateException("Kan ikke hente gjeldende vedtaksdata for blanding av uføre og alder.")
```

Comment:
Hvis man kommer hit betyr det vel bare at man har innført nye typer man ikke har tatt høyde for, ikke en blanding?

Reply:
på when() sjekker man for uføre vs alder. hvis dem ikke treffes, har man fått en blanding.